.travis/lint_04_install.sh

```diff
@@ -6,10 +6,23 @@
 
 export LC_ALL=C
 
-travis_retry pip install codespell==1.15.0
-travis_retry pip install flake8==3.5.0
-travis_retry pip install vulture==0.29
+if [ "$TRAVIS_OS_NAME" == "osx" ]; then
+  # update first to install required ruby dependency
+  travis_retry brew update
+  travis_retry brew reinstall git -- --with-pcre2 # for  --perl-regexp
```

Comment:
I don't think `--with-pcre2` is a `git` install option anymore? Or is this being passed because we're using an old version of `brew` or something?

Reply:
[This build](https://travis-ci.org/bitcoin/bitcoin/jobs/595626506) demonstrates `--with-pcre2` is needed to run `git grep --perl-regexp` in `test/link/check-doc.py`.